exchanges/graphcache/src/types.ts

```diff
@@ -146,89 +146,62 @@ export type CacheExchangeOpts = {
   storage?: StorageAdapter;
 };
 
-/**
- * The following part is meant to support the generic type-generated part of graphcache,
- * we want to make the extends loose but the default type still has to work as DataFields.
- * You can recognize these by the "generic" prefix.
- */
-type GenericResolver<
-  ParentData extends any = DataFields,
-  Args = Variables,
-  Result = ResolverResult
-> = (parent: ParentData, args: Args, cache: Cache, info: ResolveInfo) => Result;
-
-interface GenericResolverConfig {
-  [typeName: string]: {
-    [fieldName: string]: GenericResolver;
-  };
-}
-
-type GenericUpdateResolver<
-  ParentData extends any = DataFields,
-  Args = Variables
-> = (parent: ParentData, args: Args, cache: Cache, info: ResolveInfo) => void;
-
-interface GenericUpdatesConfig {
-  Mutation: {
-    [fieldName: string]: GenericUpdateResolver;
-  };
-  Subscription: {
-    [fieldName: string]: GenericUpdateResolver;
-  };
-}
-
-export type GenericCacheExchangeOpts = {
-  updates?: Partial<GenericUpdatesConfig>;
-  resolvers?: GenericResolverConfig;
-  optimistic?: OptimisticMutationConfig;
-  keys?: KeyingConfig;
-  schema?: IntrospectionData;
-  storage?: StorageAdapter;
-};
-
 // Cache resolvers are user-defined to overwrite an entity field result
 export type Resolver<
   ParentData = DataFields,
   Args = Variables,
   Result = ResolverResult
-> = (parent: ParentData, args: Args, cache: Cache, info: ResolveInfo) => Result;
-
-export interface ResolverConfig {
+> = {
+  bivarianceHack(
```

Comment:
does this ever show up in error Output? If so that could be a really awkward type trace at times 😅

Reply:
It doesn't seem to 😅